.github/workflows/flink-ci-workflow.yml

```diff
@@ -0,0 +1,77 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+
+# Workflow that is intended to be called from the main "CI Orchestrator"
+# We do this to ensure that:
+#  - all workflows that are launched by a PR will fail fast if one of them fails
+#  - all duplicate workflow runs are cancelled (i.e. only run workflows for the most recent push).
+name: "Flink CI Workflow"
+on:
+  workflow_call:
+    paths-ignore:
+      - '.github/workflows/python-ci.yml'
+      - '.github/workflows/spark-ci.yml'
+      - '.github/workflows/hive-ci.yml'
+      - '.gitignore'
+      - 'dev/**'
+      - 'mr/**'
+      - 'hive3/**'
+      - 'hive3-orc-bundle/**'
+      - 'hive-runtime/**'
+      - 'spark/**'
+      - 'pig/**'
+      - 'python/**'
+      - 'python_legacy/**'
+      - 'site/**'
+    inputs:
+      flink:
+        required: true
+        type: string
+      jvm:
+        required: true
+        type: number
```

Comment:
This part isn't needed if we keep the strategy matrix for each test suite within its own file, which would simplify the orchestrator workflow.